### DISCUSSION TITLE
can.exceptions.canoperationerror: failed to transmit: no buffer space available

### DISCUSSION THREAD

**Discussion by walker3354** [canopen-python/canopen#390] [Q&A]
Hi I am working on some can project recently,it was a little device (MCU) receiveing the terminal PDO, it works fine in the few hours but i find out that I can't transmit any PDO to my device (terminal still can receive the PDO from device) andit won't show any error message but if we restart the program it shows this message

![image.png](https://github.com/christiansandberg/canopen/assets/119646593/16c20383-e3a0-45b3-9f98-96548f433fdf)

**Comment by acolomb** [comment]
You can try experimenting with the `txqueuelen` parameter on the SocketCAN interface. I've had success to increase reliability on a Raspberry Pi by setting it to higher values like 500 or 1000.
